### DISCUSSION TITLE
v4 release date

### DISCUSSION THREAD

**Discussion by ookamidev** [lavalink-devs/Lavalink#825] [Q&A] [answered]
Hey, I'm working on a new Node.js Lavalink client library and I want to ask when the v4 should be released (if there is any release date)

I also have a second question. I have been trying to create a new player using the `/v3/sessions/sessionId/players/guildId` endpoint and I have been running into issues with Authorization. I have been sending the Node password in a "Authorization" header, is there a different way to authenticate or do I need to set a special password for the API in the config?

**Comment by topi314** [comment, marked as answer]
We currently have no release date set for `v4` as it awaits a review.
All changes are ready to be used already, however there may be bugs. Feel free to report anything not matching the docs or not working.


As for `/v3/sessions/sessionId/players/guildId` you need to set a password in your `application.yml` and provide this with the `Authorization` header. See [here](https://github.com/freyacodes/Lavalink/blob/master/IMPLEMENTATION.md#rest-api).
If you feel like you provide the correct password double check that your `application.yml` is actually taking effect (port should be 2333 and not 8080 in the logs when using the default `application.yml`)

**Reply by ookamidev** [reply]
> As for /v3/sessions/sessionId/players/guildId you need to set a password in your application.yml and provide this with the Authorization header. See [here](https://github.com/freyacodes/Lavalink/blob/master/IMPLEMENTATION.md#rest-api).
If you feel like you provide the correct password double check that your application.yml is actually taking effect (port should be 2333 and not 8080 in the logs when using the default application.yml)

I have probably misunderstood the authorization part.
in `application.yml` I have set the "password" under "server" `testPassword` for example
and now if I do
```ts
axios.patch(`http://ip:port/v3/sessions/sessionId/players/guildId?noReplace=true`, {
      Headers: {
        Authorization: 'testPassword'
      }
})
```
I receive 401 unauthorized response.
And yes, I have double checked and the password is correct.

**Reply by topi314** [reply]
The yaml path for the password needs to be `lavalink.server.password`.
Can you post your startup logs?

**Reply by ookamidev** [reply]
Actually nevermind. I have just realized it was my mistake, I was sending the Authorization in the data object instead of config object. Sorry for wasting your time!
